### ISSUE TITLE
The dataset does not contain attribute faces.

### ISSUE THREAD

**Issue by YuRongHao2233** [leolyliu/CORE4D-Instructions#2]
When I want to visualize the human-object-motion, it hints me there is no attribute faces in the file person1_poses.npz

**Comment by OctopusWen** [comment]
@YuRongHao2233 
I'm sorry. However, since we used SMPLX to store the human pose, the "faces" attribute is not provided. You can use SMPLX inference to get the faces. More details can be found in the docs/file_definitions.md.

**Comment by YuRongHao2233** [comment]
So do I need to replace the faces in your visualization code with the faces provided in this code?
@OctopusWen

**Comment by YuRongHao2233** [comment]
And I found that some data objects are not synchronized with people's movements, and some objects such as (board007_m.obj) lack faces

**Comment by YuRongHao2233** [comment]
ModuleNotFoundError: No module named 'data_processing'，and when I try to run forcasting task, I found there is a lack of module, I can not find it in your code.

**Comment by leolyliu** [comment]
> ModuleNotFoundError: No module named 'data_processing'，and when I try to run forcasting task, I found there is a lack of module, I can not find it in your code.

Hi YuRongHao2233:

Thank you for pointing out this issue. We have uploaded the ```data_processing``` module under ```./benchmarks/data_processing```. Please refer to ```./benchmarks/data_processing/README.md``` to install the environment.

We will clean up the benchmark code and continuously fix the potential bugs in its environment setup. Nevertheless, the algorithms in the code are aligned with those described in our arxiv paper.

Best

**Comment by leolyliu** [comment]
> And I found that some data objects are not synchronized with people's movements, and some objects such as (board007_m.obj) lack faces

(1) For the synchronization problem, could you please provide the sequence names of the data?

(2) For the problem of object models, several board models are broken before, we have fixed them and re-uploaded the ```object_model.zip``` on [OneDrive](https://1drv.ms/f/s!Ap-t7dLl7BFUmHl9Une1E6FLsS4J?e=RLt0Fk). Besides, we will upload high-precision object models soon.

**Comment by leolyliu** [comment]
> So do I need to replace the faces in your visualization code with the faces provided in this code? @OctopusWen

Yes, exactly. We have updated ```visualize_human_object_motion.py``` and its usage. Thank you for the feedback.

**Comment by YuRongHao2233** [comment]
`ModuleNotFoundError: dataset_statistics`，and when I try to run forcasting task, I found there is a lack of module, I can not find it in your code, too. And for some .py file, there are some bugs in them, I have fixed them, now I wish you upload the  `dataset_statistics`

**Comment by YuRongHao2233** [comment]
> > And I found that some data objects are not synchronized with people's movements, and some objects such as (board007_m.obj) lack faces
> 
> (1) For the synchronization problem, could you please provide the sequence names of the data?
> 
> (2) For the problem of object models, several board models are broken before, we have fixed them and re-uploaded the `object_model.zip` on [OneDrive](https://1drv.ms/f/s!Ap-t7dLl7BFUmHl9Une1E6FLsS4J?e=RLt0Fk). Besides, we will upload high-precision object models soon.

Of course, I will give you the name of anomaly files I find.

**Comment by leolyliu** [comment]
> `ModuleNotFoundError: dataset_statistics`，and when I try to run forcasting task, I found there is a lack of module, I can not find it in your code, too. And for some .py file, there are some bugs in them, I have fixed them, now I wish you upload the `dataset_statistics`

We have uploaded ```dataset_statistics```, including a train-test split. Since the benchmark algorithms are evaluated on our preprocessed data ("<sequence_name>/data.npz"), and the preprocessed data is too large to upload, we will modify the data processing modules to fit the current dataset.

**Comment by YuRongHao2233** [comment]
So how do I run the benchmark on an existing dataset now?

**Comment by YuRongHao2233** [comment]
So can I run a benchmark on a synthetic dataset

**Comment by leolyliu** [comment]
> So can I run a benchmark on a synthetic dataset

Yes. Before running the motion forecasting and synthesis benchmarks, please modify ```benchmarks/motion_forecasting/MDM_Interdiff/interdiff/data/prepare_hho.py``` to align with the current data formats and run it. This code can generate ```<sequence_name>/data.npz```. Or please wait up to two days to let us fix it. Sorry for the data format misalignment. The CORE4D-Synthetic shares similar formats with CORE4D-Real.

**Comment by leolyliu** [comment]
The benchmark code has been updated. We have fixed the data formatting issues.

> > So can I run a benchmark on a synthetic dataset
> 
> Yes. Before running the motion forecasting and synthesis benchmarks, please modify `benchmarks/motion_forecasting/MDM_Interdiff/interdiff/data/prepare_hho.py` to align with the current data formats and run it. This code can generate `<sequence_name>/data.npz`. Or please wait up to two days to let us fix it. Sorry for the data format misalignment. The CORE4D-Synthetic shares similar formats with CORE4D-Real.

**Comment by YuRongHao2233** [comment]
> > And I found that some data objects are not synchronized with people's movements, and some objects such as (board007_m.obj) lack faces
> 
> (1) For the synchronization problem, could you please provide the sequence names of the data?
> 
> (2) For the problem of object models, several board models are broken before, we have fixed them and re-uploaded the `object_model.zip` on [OneDrive](https://1drv.ms/f/s!Ap-t7dLl7BFUmHl9Une1E6FLsS4J?e=RLt0Fk). Besides, we will upload high-precision object models soon.

20231002, 055 and 028 , they are  obvious. And maybe other sequences are not synchronized

**Comment by YuRongHao2233** [comment]
How do I reconcile the data of ego with the named file of human-motion?

**Comment by OctopusWen** [comment]
> How do I reconcile the data of ego with the named file of human-motion?

Hi @YuRongHao2233 ,
The timestamp of ego is provided in *.mp3 files coded in [LTC format](https://en.wikipedia.org/wiki/Linear_timecode). And the timestamp of human-motion is supplied by `timestamp.txt` at the corresponding allocentric_RGBD_videos.

**Comment by YuRongHao2233** [comment]
> > How do I reconcile the data of ego with the named file of human-motion?
> 
> Hi @YuRongHao2233 , The timestamp of ego is provided in *.mp3 files coded in [LTC format](https://en.wikipedia.org/wiki/Linear_timecode). And the timestamp of human-motion is supplied by `timestamp.txt` at the corresponding allocentric_RGBD_videos.

Hi，
I mean how to match human-motion files to egocentric videos, I can't tell how they correspond at the moment, and I hope you can help me

**Comment by irem-zeynep** [comment]
Hello @leolyliu  @YuRongHao2233, how can I download CORE4D-Synthetic dataset? Drivelink contains only CORE4D-Real. Thank you for your help.

**Comment by leolyliu** [comment]
> Hello [@leolyliu](https://github.com/leolyliu) [@YuRongHao2233](https://github.com/YuRongHao2233), how can I download CORE4D-Synthetic dataset? Drivelink contains only CORE4D-Real. Thank you for your help.

@irem-zeynep Hi irem-zeynep,
We found some issues in the previous release of CORE4D-Synthetic and removed it. We are planning to release it in 2025/4.
